src/DynamoCoreWpf/ViewModels/Core/InPortViewModel.cs

```diff
@@ -276,8 +288,16 @@ private void RefreshPortDefaultValueMarkerVisible()
         /// </summary>
         protected override void RefreshPortColors()
         {
-            //Is in function state
-            if (node.NodeModel.IsPartiallyApplied)
+            //Boolean that defines if the node has a default value from the cache value
+            isReturningDefaultValue = !string.IsNullOrEmpty(node.NodeModel.CachedValue?.StringData);
```

Comment:
I don't understand this logic, many nodes return strings?

Reply:
Instead of getting the info of a Function State from isPartiallyApplied, I'm getting the info from the CachedValue because It's what is appearing in the little console below the node. The idea was to identify if the node has a result, even though It's a null result. 
Another thing is that I had to differentiate between the output 'Function' (in which the inputs have to change the color between red/blue and the output grey) and other ones (inputs always blue and output disabled). 
I would love to hear any other suggestions from you guys!
Thanks.

Reply:
I don't think we need this flag `isReturningDefaultValue`.